pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -125,19 +133,26 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, fmt.Errorf("cannot read table-of-contents: %w", err)
 	}
 
-	r.symbols, err = streamindex.NewSymbols(r.factory, r.indexVersion, int(r.toc.Symbols), cfg.VerifyOnLoad)
-	if err != nil {
-		return nil, fmt.Errorf("cannot load symbols: %w", err)
+	// Unmarshal samples from disk.
```

Comment:
Seems like this feature should be behind a configuration setting since we're putting it directly in the read path. We need to be able to disable this while testing if something isn't working.